idunn/blocks/wikipedia.py

```diff
@@ -5,7 +5,63 @@
 from requests.exceptions import HTTPError, RequestException, Timeout
 import pybreaker
 from elasticsearch import Elasticsearch
+from redis import ConnectionPool, ConnectionError, TimeoutError
 
+from redis_rate_limit import RateLimiter, TooManyRequests
+
+class HTTPError40X(HTTPError):
+    pass
+
+class WikipediaLimiter:
+    _limiter = None
+    _redis_url = None
+    _redis_db = None
+
+    @classmethod
+    def init_limiter(cls):
+        from app import settings
+
+        cls._redis_url = settings['WIKI_API_REDIS_URL']
+
+        if cls._redis_url is not None:
+            """
+            If a redis url has been provided to Idunn,
+            then we use the corresponding redis
+            service in the rate limiter.
+            """
+            url = cls._redis_url.split(":")[0]
+            port = cls._redis_url.split(":")[1]
+
+            cls._redis_db = settings['WIKI_API_REDIS_DB']
+
+            cls._limiter = RateLimiter(
+                resource='WikipediaAPI',
+                max_requests=settings['WIKI_API_RL_MAX_CALLS'],
+                expire=settings['WIKI_API_RL_PERIOD'],
+                redis_pool=ConnectionPool(host=url, port=port, db=cls._redis_db, socket_timeout=1)
+            )
+        else:
+            cls._limiter = False
+
+    @classmethod
+    def request(cls, f):
+        def wrapper(*args, **kwargs):
+            if cls._limiter is None:
+                cls.init_limiter()
+            if cls._limiter is not False:
+                """
+                We use the RateLimiter since
+                the redis service url has been provided
+                """
+                with cls._limiter.limit(client="Idunn"):
+                    return f(*args, **kwargs)
+            else:
```

Comment:
minor: you don't need this `else` nesting

Reply:
done